dns/dns_updater.go

```diff
@@ -161,6 +167,24 @@ func (u *updater) Update(update controller.IngressUpdate) error {
 	return nil
 }
 
+func (u *updater) filterUnassociated(aRecords []*route53.ResourceRecordSet) []*route53.ResourceRecordSet {
+	managedLBs := make(map[string]bool)
+	for _, dns := range u.schemeToDNS {
+		managedLBs[dns.dnsName] = true
+	}
+	filtered := make([]*route53.ResourceRecordSet, 0)
+	for _, rec := range aRecords {
+		if rec.AliasTarget != nil && rec.AliasTarget.DNSName != nil && managedLBs[*rec.AliasTarget.DNSName] {
+			log.Debugf("Keeping managed A record %s", *rec.Name)
+			filtered = append(filtered, rec)
+		} else {
+			log.Infof("Filtering non-managed A record %s", *rec.Name)
```

Comment:
I'm happy to make these changes.  I would like to ensure that I understand something first to make sure I haven't made some sort of logical error that I'm missing.  This method is filtering the results of `aRecords, err := u.r53.GetARecords()` which should only return A records (as long as my reading of the code is accurate) which is why the messages are as they are.  Are you pointing out that within the context of the `filterUnassociated` method there's nothing that guarantees that they are actually A records or have I missed something larger?